parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2017,7 +2026,10 @@ array_premature_end: eof
                    | kELSE
                    | kWHEN
                    | kIN
-                   | tRPAREN
+   premature_rbrack: tRPAREN
+                   | tRCURLY
+                   | tCOLON
```

Comment:
It's possible that I am missing something, but the intuition for why colon is a premature ending here is not obvious to me...and I'm not sure that I see any tests that try to hit that case -- though it's possible I lack the imagination to see how the current tests work with colons.  (OTOH, we seem to have had it for `array_premature_end` and it's worked out OK?)

I guess it's something like:

```ruby
T.any(String, Integer
::Some::Global::Value.call(...)
```

or something like that?

Can you explain the rationale and/or see if tests can be added for that case?

Reply:
Perhaps colons should not be included as premature ends at all?  It seems like we'd only want premature ends to be things that we know can't start another token or expression.

That one example does not seem like reason to abandon the whole approach.

Reply:
Another example: `f[end]` parses, but `[end]` does not (because we only do the strategy in #5273 for arrays, not square bracket function calls).

If we landed this change, ~all properly matched parens that have any of these tokens inside them will change from parsing as `<ErrorNode>` and the properly matching the parens, to failing to parse.